R/add_time.R

```diff
@@ -68,27 +68,82 @@ set_clin <- function(gamma, e_itv, CCOD, CCOD_t, etaC, etaE, d_itv) {
 #' S4 Class for setting parameters for time-to-events
 #'
 #' @keywords class
-.eventClass = setClass(".eventClass", slots = list(event = "character",
-                                                   lambdaC = "numeric", shape = "numericORNULL", t_itv = "numericORNULL",
-                                                   beta = "numericORNULL",
-                                                   change = "listORNULL", keep = "charORNULL"))
+.eventClass = setClass(".eventClass", slots = list(
+  event = "character",
+  lambdaC = "numeric", shape = "numericORNULL", t_itv = "numericORNULL",
+  beta = "numericORNULL",
+  change = "listORNULL", keep = "charORNULL"
+))
 
 
 
 #' Set up time-to-events
 #'
-#' @param event Distribution of time-to-events: \code{event = "pwexp"} for piece-wise exponential distribution. \code{event = "weibull"} for weibull distribution
-#' @param lambdaC Baseline hazard rate of internal control arm. Specify a vector for piece-wise hazard with duration specified in \code{t_itv} if \code{event = "pwexp"}
-#' @param beta covariates' coefficients. \code{NULL} if no covariates are found in \code{ssObj}
-#' @param shape shape of weibull distribution if \code{event = "weibull"}. \code{NULL} if \code{event = "pwexp"}
-#' @param t_itv a vector indicating interval lengths where the exponential rates provided in \code{lambdaC} apply. Note that the length of \code{t_itv} is at least 1 less than that of \code{lambdaC} and that the final value rate in \code{lambdaC} applies after time `sum(t_itv)`. \code{NULL} if \code{event = "weibull"}
-#' @param change operations applied to covariates for simulating time-to-events
-#' @param keep original covariates to keep when simulate time-to-events
+#' @description
+#'
+#' Defines the model formula and distribution to be used when simulating time-to-events. Please
+#' see the user-guide for the model formulations
+#'
+#' @param event Distribution of time-to-events: \code{event = "pwexp"} for piece-wise exponential
+#' distribution. \code{event = "weibull"} for Weibull distribution
+#'
+#' @param lambdaC Baseline hazard rate of internal control arm. Specify a vector for piece-wise
+#' hazard with duration specified in \code{t_itv} if \code{event = "pwexp"}
+#'
+#' @param beta covariates' coefficients. Must be equal in length to the number of covariates
```

Comment:
@gowerc I wonder if it's worth clarifying that the `beta` are not exponentiated? The user guide uses values that are all reasonable for HRs, and the default value of 1 I think could also be interpreted that way.